### PR TITLE
[WIP] Data Model

### PR DESCRIPTION
**Pull request by jwei98**

### CODE REVIEW

**Review by rishi-tripathy — APPROVED** [review]

File: Models.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+"""
+Define all data models.
+"""
+
+from py2neo.ogm import (GraphObject, Property, Related, RelatedTo, RelatedFrom)
+
+
+class Person(GraphObject):
+    __primarykey__ = "uid"
```
Comment (jwei98) [inline Models.py]: I'm not sure we need this because uniqueness will be determined through username.

File: Models.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    InvitedTo = Related("Event", "INVITED_TO")
+
+    def __init__(self, username, display_name, email, photo):
+        self.username = username
+        self.display_name = display_name
+        self.email = email
+        self.photo = photo
+
+class Circle(GraphObject):
+    __primarykey__ = "cid"
```
Comment (jwei98) [inline Models.py]: If we leave this line out, an automatic __id__ property will be set: https://py2neo.org/v3/ogm.html#related-objects.

Also just an fyi, "uid" stands for unique identifier, so you would never really say "cid" or "eid"

File: Models.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    name = Property()
+    cid = Property()
+
+    # Relationships.
+    Contains = Related("Person", "CONTAINS")
+    Scheduled = Related("Event", "SCHEDULED")
+
+    def __init__(self, name, members, events):
+        self.name = name
+        self.cid = 000  # How to determine this?
```
Comment (jwei98) [inline Models.py]: See above comment. We shouldn't be managing unique id's ourselves, let's leave it to neo4j

File: Models.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        # Initialize ID and Neo relationships
+
+class Event(GraphObject):
+    __primarykey__ = "eid"
```
Comment (jwei98) [inline Models.py]: Same comment as Circles primary key

File: Models.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    description = Property()
+    datetime = Property()
+
+    # Relationships.
+    InvitedTo = RelatedFrom("Person", "INVITED_TO")
+    BelongsTo = Related("Circle", "BELONGS_TO")
+
+    def __init__(self, name, description, time, circle):
+        self.name = name
+        self.eid = 000  # How to determine this?
```
Comment (jwei98) [inline Models.py]: rm. Same as above

File: Models.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        self.description = description
+        self.datetime = time
+        # is there a way to initialize the neo relationships
```
Comment (jwei98) [inline Models.py]: I don't think we need to initialize with relationships. Let's create the subject node itself first, and then add relationships in separate steps.

**Pull request merged**